app/src/main/java/rectangledbmi/com/pittsburghrealtimetracker/ui/selection/NavigationDrawerFragment.kt

```diff
@@ -0,0 +1,436 @@
+            return routes?.get(position)?.route.hashCode().toLong()
+        }
+
+        inner class BusRouteHolder(itemView: View) : RecyclerView.ViewHolder(itemView), View.OnClickListener {
+            private var mRoute: Route? = null
+            private val routeIcon: TextView
+            private val routeDescription: TextView
+            override fun onClick(v: View) {
+                if (mRoute == null) return
+                if (mRoute?.isSelected == false && selectedRoutes?.size == resources.getInteger(R.integer.max_checked)) {
+                    Toast.makeText(activity, getString(R.string.max_selected_routes), Toast.LENGTH_SHORT).show()
+                    return
+                }
+                toggleRoute(mRoute)
+                notifyItemChanged(adapterPosition)
+            }
+
+            private fun generateIcon() {
+                mRoute?.let {mRoute ->
+                    routeIcon.setBackgroundColor(mRoute.routeColor)
+                    routeIcon.text = mRoute.route
+                    routeIcon.setTextColor(if (isLight(mRoute.routeColor)) Color.BLACK else Color.WHITE)
+                }
+            }
+
+            fun bindBusRoute(busRoute: Route) {
+                mRoute = busRoute
+                mRoute?.let{mRoute ->
+                    routeDescription.text = busRoute.routeInfo
+                    generateIcon()
+                    itemView.isActivated = mRoute.isSelected
+                }
+            }
+
+            /**
+             * Decides whether or not the color (background color) is light or not.
+             *
+             *
+             * Formula was taken from here:
+             * http://stackoverflow.com/questions/24260853/check-if-color-is-dark-or-light-in-android
+             *
+             * @param color the background color being fed
+             * @return whether or not the background color is light or not (.345 is the current threshold)
+             * @since 47
+             */
+            private fun isLight(color: Int): Boolean {
+                return 1.0 - (0.299 * Color.red(color) + 0.587 * Color.green(color) + 0.114 * Color.blue(color)) / 255 < .5
+            }
+
+            init {
+                itemView.setOnClickListener(this)
+                routeDescription = itemView.findViewById<View>(R.id.bus_route_text) as TextView
+                routeIcon = itemView.findViewById<View>(R.id.bus_route_icon) as TextView
+            }
+        }
+
+        init {
+            createRoutes()
+        }
+    }
+
+    val selectedRoutesObservable: Flowable<Set<String?>?>?
+        get() = routeSelectionPublishSubject
+                ?.toFlowable(BackpressureStrategy.BUFFER)
+                ?.map(RouteSelection::selectedRoutes)
+
+    val toggledRouteObservable: Flowable<Route>?
+        get() = routeSelectionPublishSubject
+                ?.toFlowable(BackpressureStrategy.BUFFER)
+                ?.map(RouteSelection::toggledRoute)
```

Comment:
do we need `?`s here?

Reply:
the other option would be `lateinit` for the routeSelectionPublishSubject I guess, but this seems like the better way to me